include/i2c.h

```diff
@@ -395,13 +421,49 @@ static inline int i2c_read(struct device *dev, u8_t *buf,
 	return i2c_transfer(dev, &msg, 1, addr);
 }
 
+/**
+ * @brief Write then read data from an I2C device.
+ *
+ * This supports the common operation "this is what I want", "now give
+ * it to me" transaction pair through a combined write-then-read bus
+ * transaction.
+ *
+ * @param dev Pointer to the device structure for the driver instance
+ * @param write_buf Pointer to the data to be written
+ * @param num_write Number of bytes to write
+ * @param read_buf Pointer to storage for read data
+ * @param num_read Number of bytes to read
+ * @param addr Address of the I2C device
+ *
+ * @retval 0 if successful
+ * @retval negative on error.
+ */
+static inline int i2c_write_read(struct device *dev,
+				 const void *write_buf, size_t num_write,
+				 void *read_buf, size_t num_read,
+				 u16_t addr)
```

Comment:
I propose to have I2C device address as a second parameter. That's more natural. The function will read better. 

Functions which we want to keep and which 
- put dev_addr at the end are: `i2c_write`, `i2c_read`
- put dev_addr as the second parameter are: `i2c_reg_read_byte`, `i2c_reg_write_byte`, `i2c_reg_update_byte`

Either way the API will not be consistent. `i2c_write_read` will become a 'workhorse' of this API so it's nice to have it clean signature.

Reply:
I prefer to have `i2c_write_read` be consistent with `i2c_write` and `i2c_read`.

Reply:
I agree with @mnkp, device address as a second parameter is more beautiful and understandable, I would not write it like that because of i2c_write and i2c_read.

Reply:
@Mierunski @anangl @MaureenHelm your preference on this question?

I2C device address always should have immediately followed dev.  Unless I hear by mid-day Wednesday that somebody else will block the merge if the argument order is inconsistent with `i2c_write()` and `i2c_read()` I'll make the change.

Reply:
> @Mierunski @anangl @MaureenHelm your preference on this question?
> 
> I2C device address always should have immediately followed dev.

I support @mnkp's proposal. With I2C device address as the second parameter, this function will look much better.

Reply:
agree with @mnkp

Reply:
API change has been made.